pkg/reconciler/adapter.go

```diff
@@ -260,11 +261,17 @@ func commonAdapterKnServiceOptions(rcl v1alpha1.Reconcilable) []resource.ObjectO
 
 // newServiceAccount returns a ServiceAccount object with its OwnerReferences
 // metadata attribute populated from the given owners.
-func newServiceAccount(rcl v1alpha1.Reconcilable, owners []kmeta.OwnerRefable) *corev1.ServiceAccount {
-	return resource.NewServiceAccount(rcl.GetNamespace(), ServiceAccountName(rcl),
+func newServiceAccount(ctx context.Context, rcl v1alpha1.Reconcilable, owners []kmeta.OwnerRefable) *corev1.ServiceAccount {
+	extraAnnotations := v1alpha1.ServiceAccountAnnotationsFromContext(ctx)
+
+	resource := resource.NewServiceAccount(rcl.GetNamespace(), ServiceAccountName(rcl),
 		resource.Owners(owners...),
 		resource.Labels(CommonObjectLabels(rcl)),
 	)
+
+	resource.Annotations = extraAnnotations
+
+	return resource
```

Comment:
The package and the var is `called` resource, can you change the var to something like `res`.
Also, it would be nice if the ServiceAccount is fully created with options, including the annotations.

I haven't tested this, but you get the idea:

```suggestion
      opts := []resource.ObjectOption{}
       for k, v := range extraAnnotations {
		opts = append(opts, resource.Annotation(k,v)
       }
	res := resource.NewServiceAccount(rcl.GetNamespace(), ServiceAccountName(rcl),
		resource.Owners(owners...),
		resource.Labels(CommonObjectLabels(rcl)),
		opts
	)

	return res
```